typing/env.ml

```diff
@@ -450,14 +450,20 @@ type type_descriptions =
 
 let in_signature_flag = 0x01
 
+type 'a value_or_persistent =
+  | Value of 'a
+  | Persistent
```

Comment:
`Persistent` is just a placeholder for remembering that the module is an external unit.  Wouldn't it make sense to keep the actual file name, as discovered during the initial scan?

Reply:
Only summaries are serialized, so this would be fine I guess, but it requires a bit more refactoring.

Reply:
Looking at it, there is `t` in `module_components`, however it is always `empty` when saving a cmi file. I guess we could make it an option, then we would have to bootstrap less often.